civiclens/nlp/topics.py

```diff
@@ -222,7 +213,87 @@ def label_topics(topics: dict[int, list], model: TopicChain) -> dict[int, str]:
         Dictionary of topics, and labels
     """
     labels = {}
-    for topic, terms in topics.values():
+    for topic, terms in topics.items():
         labels[topic] = model.generate_label(terms)
 
     return labels
+
+
+def topic_comment_analysis(
+    comment_data: RepComments,
+    model: TopicModel = None,
+    labeler: LabelChain = None,
+    sentiment_analyzer: Callable = None,
+) -> RepComments:
+    """
+    Run topic and sentiment analysis.
+    """
+    comments: list[Comment] = []
+
+    if comment_data.summary:
+        comments += [
+            Comment(text=comment_data.summary, id="Summary", source="Summary")
+        ]
+    try:
+        comments += comment_data.to_list()
+        comment_topics = model.run_model(comments)
+    except TooFewTopics:
+        comments += comment_data.get_nonrepresentative_comments()
+        comment_topics = model.run_model(comments)
+    except Exception:
+        # log error
+        return comment_data
+
+    topic_terms = model.get_terms()
+    topic_labels = label_topics(topic_terms, labeler)
+
+    # filter out non_rep comments
+    rep_comments: list[Comment] = []
+
+    for comment in comments:
+        comment.topic_label = topic_labels[comment_topics[comment.id]]
+        comment.topic = comment_topics[comment.id]
+        comment.sentiment = sentiment_analyzer(comment)
+        if comment.representative:
+            rep_comments.append(comment)
+
+    rep_comments = sorted(
+        rep_comments, key=lambda comment: comment.num_represented, reverse=True
+    )
+
+    return RepComments(
```

Comment:
could just edit the comment_data object to include the topic info instead of creating a new object in the return statement

Reply:
Yeah, I went back and forth on that if its better to modify in place or create a new object, might ask a TA/James what's best practice

Reply:
sounds good, will resolve and push that to backlog lol